src/gc/impl/conservative/gc.d

```diff
@@ -260,6 +294,7 @@ class ConservativeGC : GC
     import core.internal.spinlock;
     static gcLock = shared(AlignedSpinLock)(SpinLock.Contention.lengthy);
     static bool _inFinalizer;
+    static bool isPrecise=false;
```

Comment:
After stumbling over a number of unrelated issues (I thought these days are gone) I figured the issue: `isPrecise` must be `__gshared`, otherwise stuff allocated in other threads won't get correct pointer information and GC collections triggered outside the main thread are done conservatively!

Long term, these statics should be members instead, probably by merging Gcx into ConservativeGC. Why are these separated to begin with?